abnf/odata-abnf-testcases.yaml

```diff
@@ -2268,6 +2272,13 @@ TestCases:
     Rule: odataRelativeUri
     Input: Customers?$select=Addresses(@a=$this;$filter=endswith(@a/Street,'gasse'))
 
+  - Name: request reference in $batch
+    Rule: commonExpr
+    Input: Customer eq $1
+    Expect:
+      - firstMemberExpr:Customer
+      - firstMemberExpr:$1
+
```

Comment:
Add examples for request reference followed by memberExpr and collectionNavigationExpr